src/components/data/FileBrowser.js

```diff
@@ -164,57 +179,109 @@ export const FileBrowserPanel = _.flow(
-            { header: h(HeaderCell, ['Size']), size: { basis: 200, grow: 0 }, key: 'size' },
-            { header: h(HeaderCell, ['Last modified']), size: { basis: 200, grow: 0 }, key: 'updated' }
-          ],
-          rows: [
-            ..._.map(p => {
-              return {
-                name: h(TextCell, [
-                  makeBucketLink({
-                    label: getBareFilename(p),
-                    target: getFullPrefix(p),
-                    onClick: () => setPrefix(p)
-                  })
-                ])
-              }
-            }, prefixes),
-            ..._.map(({ name, size, updated }) => {
-              return {
-                button: h(Link, {
-                  style: { display: 'flex' },
-                  onClick: () => setDeletingName(name),
-                  tooltip: 'Delete file'
-                }, [
-                  icon('trash', { size: 16, className: 'hover-only' })
-                ]),
-                name: h(TextCell, [
-                  makeBucketLink({
-                    label: getBareFilename(name),
-                    target: name,
-                    onClick: () => setViewingName(name)
-                  })
-                ]),
-                size: filesize(size, { round: 0 }),
-                updated: Utils.makePrettyDate(updated)
+        'Drag and drop your files here'
+      ]),
+      div({
+        style: { fontSize: '1rem', flex: '1 1 auto', padding: '1em', height: '100%' }
+      }, [
+        h(AutoSizer, {}, [
+          ({ width, height }) => h(GridTable, {
+            ref: table,
+            width,
+            height,
+            rowCount: numPrefixes + numObjects,
+            noContentMessage: 'No files have been uploaded yet',
+            onScroll: saveScroll,
+            initialX,
+            initialY,
+            rowHeight: 40,
+            headerHeight: 40,
+            styleCell: () => {
+              return { padding: '0.5em', borderRight: 'none', borderLeft: 'none' }
+            },
+            styleHeader: () => {
+              return { padding: '0.5em', borderRight: 'none', borderLeft: 'none' }
+            },
+            columns: [
+              {
+                width: 30,
+                headerRenderer: () => '',
+                cellRenderer: ({ rowIndex }) => {
+                  if (rowIndex >= numPrefixes) {
+                    const { name } = objects[rowIndex - numPrefixes]
+                    return h(Link, {
+                      style: { display: 'flex' },
+                      onClick: () => setDeletingName(name),
+                      tooltip: 'Delete file'
+                    }, [
+                      icon('trash', {
+                        size: 16,
+                        className: 'hover-only'
+                      })
+                    ])
+                  }
+                }
+              },
+              {
+                width: width - 400, // Fill the remaining space
```

Comment:
Just a preliminary note: If this table isn't intended to scroll horizontally, the `FlexTable` component might be a better fit, since it will give you virtualized rows and you can specify flexbox-like constraints for the column sizes.

Reply:
Ah, thanks!  Yeah, I saw that one but hadn't looked much into it.

Happy to switch to that one before this PR gets merged, or come back and fix it later. Whichever you'd prefer...

Reply:
Andrew, I'm going to leave it up to you. As long as the current code meets your need, I think it's fine to go ahead with it and refactor later.

Reply:
I just pushed an update to switch to a `FlexTable`. You're right, the conversion was trivial and it's definitely a cleaner implementation for this use case.

I added `rowHeight` and `headerHeight` parameters, as well as `styleCell()` and `styleHeader()` functions, in order to attempt to match the styling of File Browsers elsewhere on the site. Since these are all optional, their presence should be a no-op for anything else using these tables.